core/src/main/java/cucumber/runtime/arquillian/feature/FeaturesManager.java

```diff
@@ -0,0 +1,370 @@
+        final String featurePath = resource.getPath();
+        final File featureFileDump = dump(tempDir, javaClass.getName() + '/' + featurePath, feature);
+        featureFileDump.deleteOnExit();
+        return featureFileDump;
+    }
+    
+    private static Map<String, Collection<URL>> extractUrlMapFromCucumberSearcher(final ClassLoader classLoader, final String rawFeatureURL, final String featureHome) throws IllegalStateException {
+    	final Map<String, Collection<URL>> featureURLMap = new HashMap<String, Collection<URL>>();    	
+    	int lineSeparatorIndex = getSeparatorIndex(rawFeatureURL);
+        final String urlPath = extractFeatureURLPath(lineSeparatorIndex, rawFeatureURL);
+        final String urlSuffix = extractFeatureURLSuffix(lineSeparatorIndex, rawFeatureURL);    	
+        featureURLMap.put(urlPath + urlSuffix, createListFromCucumberSearcher(classLoader, featureHome, urlPath));    	
+        return featureURLMap;
+    }
+    
+    private static List<URL> createListFromCucumberSearcher(final ClassLoader classLoader, final String featuresHome, final String urlPath) throws IllegalStateException {
+    	final List<URL> urlsList = new ArrayList<URL>();        
+        try {
+        	urlsList.addAll(createUrlsListFromResource(classLoader, urlPath));
+        } catch(Exception e)
+        {
+        	//do nothing
+        }
+        
+        try {
+        	urlsList.addAll(createUrlsListFromResource(classLoader, featuresHome+urlPath));
+        } catch(Exception e)
+        {
+        	//do nothing
+        }
+        
+        if(urlsList.isEmpty()) { 
+        	throw new IllegalStateException();
+        }        
+        return urlsList;
+    }
+    
+    private static List<URL> createUrlsListFromResource(final ClassLoader classLoader, final String resourcesPath) throws IllegalStateException {
+    	final List<URL> urlsList=new ArrayList<URL>();
+    	final MultiLoader multiLoader = new MultiLoader(classLoader);
+        final Iterator<Resource> resources;
+        try {
+        	resources = multiLoader.resources(resourcesPath, EXTENSION).iterator();
+        } catch (final IllegalArgumentException iae) { // not a directory...
+            throw new IllegalStateException();
+        }
+
+        while (resources.hasNext()) {
+            final Resource resource = resources.next();
+            final URL extractedURL = extractUrlFromResource(resource, classLoader);
+            
+            if(extractedURL!=null) {
+            	urlsList.add(extractedURL);
+            } else {
+            	LOGGER.warning("Resource " + resource + " ignored (unknown type).");
+            }            
+        }
+        
+        if(urlsList.isEmpty()) {
+        	throw new IllegalStateException();
+        }        
+        return urlsList;
+    }
+    
+    private static URL extractUrlFromResource(final Resource resource, final ClassLoader classLoader) {    	
```

Comment:
globally - this is particularly true here but the whole refactoring could get this comment *from me* - I don't like that much such small methods when they doesn't make sense outside their own usage.

IMO it doesn't help readability adding indirection

wdyt?